plugin-build/src/main/kotlin/io/sentry/android/gradle/autoinstall/fragment/FragmentInstallStrategy.kt

```diff
@@ -0,0 +1,57 @@
+package io.sentry.android.gradle.autoinstall.fragment
+
+import io.sentry.android.gradle.SentryPlugin
+import io.sentry.android.gradle.autoinstall.AutoInstallState
+import io.sentry.android.gradle.autoinstall.InstallStrategyRegistrar
+import io.sentry.android.gradle.autoinstall.SENTRY_GROUP
+import io.sentry.android.gradle.util.info
+import javax.inject.Inject
+import org.gradle.api.artifacts.ComponentMetadataContext
+import org.gradle.api.artifacts.ComponentMetadataRule
+import org.gradle.api.artifacts.dsl.ComponentMetadataHandler
+import org.slf4j.Logger
+
+// @Inject is needed to avoid Gradle error
+// @CacheableRule // TODO: make it cacheable somehow (probably depends on parameters)
+abstract class FragmentInstallStrategy @Inject constructor() : ComponentMetadataRule {
+
+    private var logger: Logger = SentryPlugin.logger
+
+    constructor(logger: Logger) : this() {
+        this.logger = logger
+    }
```

Comment:
Do we need a logger ctor or just for testing? otherwise no need to assign `SentryPlugin.logger` directly?

Reply:
just for testing. Not sure wdym by no need to assign directly - how'd you see it otherwise?

Reply:
its just a nit, you are assigning twice during tests, since it assigns already in the declaration, and you cannot make it a `val` but only `var`.
I'd rather have an optional parameter with a pre-assigned value, making it a `val`.

Reply:
mm, but this ctor is not gonna be called by Gradle, because it's Gradle who instantiates this class through `newInstance` using primary ctor. And all parameters in the ctor should be injected through `@Inject`, but we can't inject the logger unfortunately.

Reply:
Yep, I know it, again it's just a nit, but See the example https://github.com/getsentry/sentry-java/blob/e1a7226bead691b2a824332f3d3a0930e68e33cc/sentry-servlet/src/main/java/io/sentry/servlet/SentryServletRequestListener.java#L21-L29

There's an empty ctor, which defaults to a value and calls the other ctor with the given value, my idea would be to do the same.
Feel free to ignore if I'm not getting the point.